### PR TITLE
Migrate from TravisCI to GitHub Actions

### LINKED ISSUE #53: Migrate build script to GitHub Actions
To follow with practices in WordPress Core, the Docker image building should be moved from Travis to GitHub Actions.

### PR DESCRIPTION
**Pull request by desrosj**
This PR moves the build tasks from TravisCI to GitHub Actions.

Needed for this to work: a repository secret named `DOCKER_PASSWORD` needs to be added containing the password needed to login to the Docker hub.

Fixes #53.

### CODE REVIEW

**Comment by pento** [comment]
> Needed for this to work: a repository secret named DOCKER_PASSWORD needs to be added containing the password needed to login to the Docker hub.

I've added this, I'm assuming it just needs the same value as was in the Travis config.

File: .github/workflows/docker-hub.yml [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    needs: check-for-duplicates
+    strategy:
+      matrix:
+        php: [ '5.2', '5.3', '5.4', '5.5', '5.6.20', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
+    env:
+      PHP_VERSION: ${{ matrix.php }}
+      PHP_LATEST: '7.4'
+
+    steps:
+      - name: Cancel previous runs of this workflow (pull requests only)
```
Comment (noisysocks) [inline .github/workflows/docker-hub.yml:48]: Why does this need to be here if we have `needs: check-for-duplicates`?

File: .github/workflows/docker-hub.yml [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  push:
+    branches: master
+  # Once weekly On Sundays at 00:00 UTC.
+  schedule:
+    - cron: '0 0 * * 0'
+
+env:
+  REGISTRY_USERNAME: garypendergast
+  REGISTRY_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+  PACKAGE_REGISTRY_HOST:
```
Comment (noisysocks) [inline .github/workflows/docker-hub.yml:20]: Should this be empty?

File: .github/workflows/docker-hub.yml [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  PACKAGE_REGISTRY: wordpressdevelop
+  PR_TAG:
```
Comment (noisysocks) [inline .github/workflows/docker-hub.yml:22]: Should this be empty?

File: .github/workflows/docker-hub.yml [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  build-php-images:
+    name: PHP ${{ matrix.php }} Image
+    runs-on: ubuntu-latest
+    needs: check-for-duplicates
+    strategy:
+      matrix:
+        php: [ '5.2', '5.3', '5.4', '5.5', '5.6.20', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
+    env:
+      PHP_VERSION: ${{ matrix.php }}
+      PHP_LATEST: '7.4'
```
Comment (noisysocks) [inline .github/workflows/docker-hub.yml:45]: Not 8.0? 😀

File: .github/workflows/github-container-registry.yml [hunk 5]
```diff
@@ -0,0 +1,169 @@
+##########################################################################
+#
+# WARNING: This file was generated by update.php.
+#
+# You can find the relevant template in the `/templates` folder.
+#
+name: Build test images
+
+on:
+  pull_request:
```
Comment (noisysocks) [inline .github/workflows/github-container-registry.yml:10]: Why every PR? Maybe I don't understand what we use GitHub container repository for 🙂

File: .github/workflows/github-container-registry.yml [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    needs: check-for-duplicates
+    strategy:
+      matrix:
+        php: [ '5.2', '5.3', '5.4', '5.5', '5.6.20', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0' ]
+    env:
+      PHP_VERSION: ${{ matrix.php }}
+      PHP_LATEST: '7.4'
+
+    steps:
+      - name: Cancel previous runs of this workflow (pull requests only)
```
Comment (noisysocks) [inline .github/workflows/github-container-registry.yml:44]: No need to do this because of `needs: check-for-duplicates`.

**Review by noisysocks — COMMENTED** [review]
I looked through the generated files and left a few notes.

The proof is in whether or not this ends up working, really, so I'm happy to merge this to see what happens 🙂

File: .github/workflows/docker-hub.yml (hunk 2 above)
Comment (desrosj) [reply .github/workflows/docker-hub.yml:20]: For the Docker Hub workflow, yes. The login command defaults to using Docker Hub, so no value is needed.

File: .github/workflows/docker-hub.yml (hunk 3 above)
Comment (desrosj) [reply .github/workflows/docker-hub.yml:22]: PR tags should only be used when creating the test containers that are submitted to the GitHub container registry, so this is alright to be empty.

File: .github/workflows/docker-hub.yml (hunk 4 above)
Comment (desrosj) [reply .github/workflows/docker-hub.yml:45]: I'm going to make that change in a separate PR after this one is merged and builds are back working correctly. For now, this copies the "latest" value currently being used.

File: .github/workflows/docker-hub.yml (hunk 1 above)
Comment (desrosj) [reply .github/workflows/docker-hub.yml]: Ah, thanks for finding that. This is unnecessary. I forgot to regenerate the files after I removed that from the template.

File: .github/workflows/github-container-registry.yml (hunk 5 above)
Comment (desrosj) [reply .github/workflows/github-container-registry.yml:10]: We use GitHub container registry to host generated images for testing changes made in PRs before merging and pushing the containers out to Docker Hub. So this workflow should only run for pull requests.

File: .github/workflows/github-container-registry.yml (hunk 6 above)
Comment (desrosj) [reply .github/workflows/github-container-registry.yml]: Fixed!

**Comment by desrosj** [comment]
Looks like it worked! Thanks everyone!

**Pull request merged**